scala/data_structures/bloom_filter.scala

```diff
@@ -1,66 +1,78 @@
 package data_structures
 
+import scala.util.hashing.MurmurHash3.stringHash
+import Math.{abs, ceil, log, pow, round}
+
 case class BloomFilter[A](
   nbOfItems: Int,
   falsePositiveProbability: Float,
   private val array: Array[Boolean],
-  private val hashFunctions: Array[String => Int]
+  private val numberOfHashFunctions: Int,
+  private val hashSeed: Int
 ) {
 
+  private val hashFunctions: Stream[String => Int] =
+    (1 to numberOfHashFunctions)
+      .toStream
+      .map(i => (str: String) => abs(stringHash(str, hashSeed + i)) % array.length)
```

Comment:
This is generated upon creation of the bloom filter, but it is a lazy sequence. Hence, it won't occupy any space.

Reply:
This is also where we can see an important part of the PR: to the random seed is simply added the sequence's current index. Hence, each hash function has a distinct seed, but each seed is predictable (given we know the random seed, which a potential attacker doesn't).